pkg/test/utils/kubernetes.go

```diff
@@ -35,6 +36,60 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// ValidateErrors accepts an error as its first argument and passes it to each function in the errValidationFuncs slice,
+// if any of the methods returns true, the method returns nil, otherwise it returns the original error.
+func ValidateErrors(err error, errValidationFuncs ...func(error) bool) error {
+	for _, errFunc := range errValidationFuncs {
+		if errFunc(err) {
+			return nil
+		}
+	}
+
+	return err
+}
+
+// Retry retries a method until the context expires or the method returns an unvalidated error.
+func Retry(ctx context.Context, fn func(context.Context) error, errValidationFuncs ...func(error) bool) error {
+	var err error
+
+	// do { } while (err != nil): https://stackoverflow.com/a/32844744/10892393
+	for ok := true; ok; ok = err != nil {
+		done := make(chan bool)
+
+		// run the function in a goroutine and close it once it is finished so that
```

Comment:
This feels atrocious. Is there a better way to wait for either function return or context expiration?

Reply:
I think the only change I'd make here is having an errChan so that you're not trying to set the error across goroutine bondaries - and then add your errChan to your select.  Otherwise, yeah, it is what it is.

Additionally, would https://github.com/kubernetes/kubernetes/blob/master/pkg/util/goroutinemap/exponentialbackoff/exponential_backoff.go be useful?

Reply:
Agree with the channel fix.  I think there will be a race condition otherwise

Reply:
Okay, I think it should be fixed now. Better?